src/crypto/bn254/Fp6.zig

```diff
@@ -179,8 +180,20 @@ pub fn scalarMulAssign(self: *Fp6, scalar: *const Fp) void {
     self.* = self.scalarMul(scalar);
 }
 
+pub fn mulByFp2(self: *const Fp6, fp2_val: *const Fp2) Fp6 {
+    return Fp6{
+        .v0 = self.v0.mul(fp2_val),
+        .v1 = self.v1.mul(fp2_val),
+        .v2 = self.v2.mul(fp2_val),
+    };
+}
+
+pub fn mulByFp2Assign(self: *Fp6, fp2_val: *const Fp2) void {
+    self.* = self.mulByFp2(fp2_val);
+}
```

Comment:
nit: I would likely just inline this whenever we needed to do this rather than creating a new method